docs/src/extend/custom-rule-tutorial.md

```diff
@@ -0,0 +1,452 @@
+```
+
+## Step 2: Stub Out the Rule File
+
+In the `enforce-foo-bar.js` file, add some scaffolding for the `enforce-foo-bar` custom rule. Also, add a `meta` object with some basic information about the rule.
+
+```javascript
+// enforce-foo-bar.js
+
+module.exports = {
+    meta: {
+       // TODO: add metadata
+    },
+    create(context) {
+        return {
+            // TODO: add callback function(s)
+        };
+    }
+};
+```
+
+## Step 3: Add Rule Metadata
+
+Before writing the rule, add some metadata to the rule object. ESLint uses this information when running the rule.
+
+Start by exporting an object with a `meta` property containing the rule's metadata, such as the rule type, documentation, and fixability. In this case, the rule type is "problem," the description is "Enforce that a variable named `foo` can only be assigned a value of 'bar'.", and the rule is fixable by modifying the code.
+
+```javascript
+// enforce-foo-bar.js
+
+module.exports = {
+    meta: {
+        type: "problem",
+        docs: {
+            description: "Enforce that a variable named `foo` can only be assigned a value of 'bar'.",
+        },
+        fixable: "code"
+    },
+    create(context) {
+        return {
+            // TODO: add callback function(s)
+            }
+        };
+    }
+};
+```
+
+To learn more about rule metadata, refer to [Rule Structure](custom-rules#rule-structure).
+
+## Step 4: Add Rule Visitor Methods
+
+Define the rule's `create` function, which accepts a `context` object and returns an object with a property for each syntax node type you want to handle. In this case, you want to handle `VariableDeclaration` nodes.
+You can choose any [ESTree node type](https://github.com/estree/estree) or [selector](selectors).
+
+Inside the `VariableDeclaration` visitor method, check if the node represents a `const` variable declaration, if its name is `foo`, and if it's not assigned to the string `"bar"`. You do this by evaluating the `node` passed to the `VariableDeclaration` method.
+
+If the `const foo` declaration is assigned a value of `"bar"`, then the rule does nothing. If `const foo` **is not** assigned a value of `"bar"`, then `context.report()` reports an error to ESLint. The error report includes information about the error and how to fix it.
+
+```javascript
+// enforce-foo-bar.js
+module.exports = {
+    meta: {
+        type: "problem",
+        docs: {
+            description: "Enforce that a variable named `foo` can only be assigned a value of 'bar'.",
+        },
+         fixable: "code"
+    },
+    create(context) {
+        return {
+            // Performs action in the function on every variable declaration
+            VariableDeclaration(node) {
+                // Check if a `const` variable declaration
+                if(node.kind === "const") {
+                    // Check if variable name is `foo`
+                    if(node.declarations[0].id.name === "foo") {
```

Comment:
This checks only the first VariableDeclarator. Should the rule catch e.g. `foo = "baz"` in `const a = 1, foo = "baz";`?

Reply:
for the sake of simplicity, i think it's fine as is. but if you feel otherwise, we can refactor.